File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        leftp = self.maxDepth(root.left)
        rightp = self.maxDepth(root.right)
        return 1 + max(leftp, rightp)

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        int cmt = 0;
        int result = 0;
        dfs(root, cmt, k, result);
        return result;
    }

    void dfs(TreeNode* root, int& cmt, int k, int& result) {
        // base case, not node or count is large than k then return stack
        if (!root || cmt >= k) {
            return;
        }

        dfs(root->left, cmt, k, result);
        // when found the kth smallest
        cmt++;
        if (cmt == k) {
            result = root->val;
            return;
        }
        
        return dfs(root->right, cmt, k, result);
    }
};
